app/estimate_square_size.go

```diff
@@ -1,204 +1,90 @@
-		msgShares += msgLen
+	// assume that we have to add a lot of padding by simply doubling the number
+	// of shares used
+	//
+	// TODO: use a more precise estimation that doesn't over
+	// estimate as much
+	totalSharesUsed := uint64(txSharesUsed + blobSharesUsed)
+	totalSharesUsed *= 2
+	minSize := uint64(math.Sqrt(float64(totalSharesUsed)))
+	squareSize = shares.RoundUpPowerOfTwo(minSize)
+	if squareSize >= appconsts.MaxSquareSize {
+		squareSize = appconsts.MaxSquareSize
 	}
-
-	// calculate the smallest possible square size that could contain all the
-	// shares
-	squareSize := shares.RoundUpPowerOfTwo(int(math.Ceil(math.Sqrt(float64(txShares + msgShares)))))
-
-	// the starting square size should at least be the minimum
-	if squareSize < appconsts.MinSquareSize {
+	if squareSize <= appconsts.MinSquareSize {
 		squareSize = appconsts.MinSquareSize
 	}
 
-	var fits bool
-	for {
-		// assume that all the msgs in the square use the non-interactive
-		// default rules and see if we can fit them in the smallest starting
-		// square size. We start the cursor (share index) at the beginning of
-		// the blob shares (txShares), because shares that do not
-		// follow the non-interactive defaults are simple to estimate.
-		fits, msgShares = shares.FitsInSquare(txShares, squareSize, msgLens...)
-		switch {
-		// stop estimating if we know we can reach the max square size
-		case squareSize >= appconsts.MaxSquareSize:
-			return appconsts.MaxSquareSize, txShares + msgShares
-		// return if we've found a square size that fits all of the txs
-		case fits:
-			return uint64(squareSize), txShares + msgShares
-		// try the next largest square size if we can't fit all the txs
-		case !fits:
-			// double the square size
-			squareSize = shares.RoundUpPowerOfTwo(squareSize + 1)
-		}
-	}
+	return squareSize, txSharesUsed
 }
 
-// rawShareCount calculates the number of shares taken by all of the included
-// txs and each blob. blobLens is a slice of the number of shares used by each
-// blob without accounting for the non-interactive default rules.
-func rawShareCount(txs []*parsedTx) (txShares int, blobLens []int) {
-	// blobSummary is used to keep track of the size and the namespace so that we
-	// can sort the blobs by namespace before returning.
-	type blobSummary struct {
-		// size is the number of shares used by this blob
-		size      int
-		namespace []byte
-	}
-
-	var blobSummaries []blobSummary //nolint:prealloc
-
-	// we use bytes instead of shares for tx as they are encoded contiguously in
-	// the square, unlike blobs where each of which is assigned their own set of
-	// shares
-	txBytes := 0
-	for _, pTx := range txs {
-		// if there is no wire message in this tx, then we can simply add the
-		// bytes and move on.
-		if pTx.msg == nil {
-			txBytes += len(pTx.rawTx)
+// estimateCompactShares estimates the number of shares used by compact shares
+func estimateCompactShares(squareSize uint64, ptxs []parsedTx) int {
```

Comment:
[compact shares](https://pkg.go.dev/github.com/celestiaorg/celestia-app@v0.10.0/pkg/shares#hdr-Compact_Share_Schema) are shares that can contain multiple individual things in each one. This encoding scheme allows for increased efficiency as to what can fit in those shares. Since we've removed evidence from the square, this is only celestia-specific transactions, but could include other things in the future such as ISRs (intermediate state roots). 

This is opposed to Sparse shares, which only contain a single thing in those shares. This is how blobs are encoded.